src/key_variants/single_key.rs

```diff
@@ -25,6 +31,40 @@ fn challenge_hash(R: JubJubExtended, message: BlsScalar) -> JubJubScalar {
 
 /// An Schnorr signature, produced by signing a message with a
 /// [`SecretKey`].
+///
+/// The `Signature` struct encapsulates variables of the Schnorr scheme.
+///
+/// ## Fields
+///
+/// - `u`: The `u` scalar representing the Schnorr signature.
+/// - `R`: The `R` point produced as part of the Schnorr signature.
```

Comment:
maybe add that these are `JubJubScalar` and `JubJubPoints`. This might also be applicable in other parts of the documenation